regression_tests/utils.go

```diff
@@ -0,0 +1,324 @@
+	}
+
+	return &regressionTestConfig{
+		ClientOneURL:         os.Getenv(ClientOneURLEnvVar),
+		ClientTwoURL:         os.Getenv(ClientTwoURLEnvVar),
+		ClientOneLeaderXPriv: os.Getenv(ClientOneLeaderXPrivEnvVar),
+		ClientTwoLeaderXPriv: os.Getenv(ClientTwoLeaderXPrivEnvVar),
+	}, nil
+}
+
+// isSPVWalletRunning checks if the SPV wallet is running at the specified URL.
+func isSPVWalletRunning(url string) bool {
+	url = addPrefixIfNeeded(url)
+	resp, err := http.Get(url)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return false
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("error reading response body:", err)
+		return false
+	}
+
+	var walletResp WalletResponse
+	if err := json.Unmarshal(body, &walletResp); err != nil {
+		fmt.Println("error parsing response JSON:", err)
+		return false
+	}
+
+	return walletResp.Message == spvWalletIndexResponse
+}
+
+// addPrefixIfNeeded adds the HTTP prefix to the URL if it is missing.
+func addPrefixIfNeeded(url string) string {
+	if !isValidURL(url) {
+		return domainPrefix + url
+	}
+	return url
+}
+
+// getSharedConfig retrieves the shared configuration from the SPV Wallet.
+func getSharedConfig(xpub string) (*models.SharedConfig, error) {
+	req, err := http.NewRequest(http.MethodGet, addPrefixIfNeeded(domainLocalHost)+domainSuffixSharedConfig, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set(models.AuthHeader, xpub)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to get shared config: %s", resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var configResponse models.SharedConfig
+	if err := json.Unmarshal(body, &configResponse); err != nil {
+		return nil, err
+	}
+
+	if len(configResponse.PaymailDomains) != 1 {
+		return nil, fmt.Errorf("expected 1 paymail domain, got %d", len(configResponse.PaymailDomains))
+	}
+	return &configResponse, nil
+}
+
+// promptUserAndCheck prompts the user with a question and validates the response.
+func promptUserAndCheck(question string) (int, error) {
+	reader := bufio.NewReader(os.Stdin)
+	var response string
+	var checkResult int
+
+	for {
+		timeout := time.After(timeoutDuration)
+		input := make(chan string)
+		go func() {
+			fmt.Println(question)
+			response, _ = reader.ReadString('\n')
+			response = strings.TrimSpace(response)
+			input <- response
+		}()
+
+		select {
+		case <-timeout:
+			return -1, ErrTimeout
+		case response = <-input:
+			checkResult = checkResponse(response)
+			if checkResult != -1 {
+				return checkResult, nil
+			}
+			fmt.Println("Invalid response. Please answer y/yes or n/no.")
+		}
+	}
+}
+
+// checkResponse checks the response and returns an integer indicating the result.
+func checkResponse(response string) int {
```

Comment:
Consider some `yes`, `no`, `noChoice` enum

Reply:
Constants introduced to these values.